src/main/java/Problem3/Problem3.java

```diff
@@ -6,6 +6,16 @@
 
 public class Problem3 {
     public static void printListInReverse(ListNode list, PrintStream print) {
-        // homework
+
+        if (list != null && list.next != null) {
```

Comment:
Since you are using recursion, you only need to check one base case:
`if (list == null) {
            return;
        }
        printListInReverse(list.next, print);
        print.println(list.val);`